scripts/email.js

```diff
@@ -0,0 +1,113 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    checkEmailValidity,
+    clearFormErrors,
+    errorList,
+    disableFormFields,
+    enableFormFields,
+    postToEmailServer
+} from './form-utils';
+
+let form;
+
+const EmailForm = {
+    handleFormError: (msg) => {
+        let error;
+
+        enableFormFields(form);
+
+        form.querySelector('.mzp-c-form-errors').classList.remove('hidden');
+
+        switch (msg) {
+            case errorList.EMAIL_INVALID_ERROR:
+                error = form.querySelector('.error-email-invalid');
+                break;
+            case errorList.PRIVACY_POLICY_ERROR:
+                error = form.querySelector('.error-privacy-policy');
+                break;
+            default:
+                error = form.querySelector('.error-try-again-later');
+        }
+
+        if (error) {
+            error.classList.remove('hidden');
+        }
+    },
+
+    handleFormSuccess: () => {
+        form.classList.add('hidden');
+        const thanks = document.getElementById('newsletter-thanks');
+        thanks.style.display = "block"
+    },
+
+    validateFields: () => {
+        const email = form.querySelector('input[type="email"]').value;
+        const privacy = !!form.querySelector('input[name="privacy"]:checked');
+
+        // Really basic client side email validity check.
+        if (!checkEmailValidity(email)) {
+            EmailForm.handleFormError('Invalid email address');
+            return false;
+        }
+
+        // Confirm privacy policy is checked
+        if (!privacy) {
+        EmailForm.handleFormError('Privacy policy not checked');
+            return false;
+        }
+
+        return true;
+    },
+
+    sumbit: (e) => {
```

Comment:
I may have found the reason it wasn't POST-ing and not giving us a 200 status code.
You misspelled "submit"!
```suggestion
    submit: (e) => {
```

Reply:
The function is misspelt both here and where the event is bound, so whilst the spelling should be corrected, I’m not sure that’s the reason.

Reply:
🙃

Reply:
Nevermind :)